### PR TITLE
Nested auth cache redirect uri

### PR DESCRIPTION
**Pull request by juan-arias**
## Proposed changes

Describe what this PR is trying to do.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/cache/token/MSIDCredentialCacheItem.m [hunk 1]
```diff
@@ -51,9 +51,9 @@ @implementation MSIDCredentialCacheItem
 
 - (NSString *)description
 {
-    return [NSString stringWithFormat:@"MSIDCredentialCacheItem: clientId: %@, credentialType: %@, target: %@, realm: %@, environment: %@, expiresOn: %@, extendedExpiresOn: %@, refreshOn: %@, cachedAt: %@, last recovery attempted at: %@, familyId: %@, homeAccountId: %@, enrollmentId: %@, speInfo: %@, secret: %@",
+    return [NSString stringWithFormat:@"MSIDCredentialCacheItem: clientId: %@, credentialType: %@, target: %@, realm: %@, environment: %@, expiresOn: %@, extendedExpiresOn: %@, refreshOn: %@, cachedAt: %@, last recovery attempted at: %@, familyId: %@, homeAccountId: %@, enrollmentId: %@, speInfo: %@, secret: %@, redirectUri: %@",
```
Comment (oldalton) [inline IdentityCore/src/cache/token/MSIDCredentialCacheItem.m:54]: Let's mask redirectURI here

File: IdentityCore/src/oauth2/MSIDOauth2Factory.m [hunk 2]
```diff
@@ -262,6 +262,12 @@ - (BOOL)fillAccessToken:(MSIDAccessToken *)token
     
     token.expiresOn = [NSDate dateWithTimeIntervalSince1970:(uint64_t)[expiresOn timeIntervalSince1970]];
     token.cachedAt = [NSDate dateWithTimeIntervalSince1970:(uint64_t)[[NSDate date] timeIntervalSince1970]];
+    
+    // Cache redirect uri
```
Comment (oldalton) [inline IdentityCore/src/oauth2/MSIDOauth2Factory.m:266]: Should this live in AAD factory?

File: IdentityCore/src/oauth2/token/MSIDAccessToken.m [hunk 3]
```diff
@@ -244,8 +249,8 @@ - (void)setScopes:(NSOrderedSet<NSString *> *)scopes
 - (NSString *)description
 {
     NSString *baseDescription = [super description];
-    return [baseDescription stringByAppendingFormat:@"(access token=%@, expiresOn=%@, extendedExpiresOn=%@,refreshOn=%@, target=%@, enrollmentId=%@, applicationIdentfier=%@)",
-            [_accessToken msidSecretLoggingHash], _expiresOn, _extendedExpiresOn, _refreshOn, _target, [_enrollmentId msidSecretLoggingHash], _applicationIdentifier];
+    return [baseDescription stringByAppendingFormat:@"(access token=%@, expiresOn=%@, extendedExpiresOn=%@,refreshOn=%@, target=%@, enrollmentId=%@, applicationIdentfier=%@, redirectUri=%@)",
```
Comment (oldalton) [inline IdentityCore/src/oauth2/token/MSIDAccessToken.m:252]: Mask redirectURI here

**Review by oldalton — APPROVED** [review]

File: IdentityCore/src/cache/token/MSIDCredentialCacheItem.m (hunk 1 above)
Comment (juan-arias) [reply IdentityCore/src/cache/token/MSIDCredentialCacheItem.m:54]: Updated

File: IdentityCore/src/oauth2/token/MSIDAccessToken.m (hunk 3 above)
Comment (juan-arias) [reply IdentityCore/src/oauth2/token/MSIDAccessToken.m:252]: Updated

File: IdentityCore/src/oauth2/MSIDOauth2Factory.m (hunk 2 above)
Comment (juan-arias) [reply IdentityCore/src/oauth2/MSIDOauth2Factory.m]: Moved to `MSIDAADV2Oauth2Factory`

**Pull request merged**